richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichTextState.kt

```diff
@@ -569,21 +583,41 @@ class RichTextState internal constructor(
         )
     }
 
+    @Deprecated("Use toggleUnorderedList() instead to add or remove unordered list")
     fun removeUnorderedList() {
         val paragraph = getRichParagraphByTextIndex(selection.min - 1) ?: return
+        removeUnorderedList(paragraph)
+    }
+
+    private fun removeUnorderedList(paragraph: RichParagraph) {
         if (paragraph.type !is UnorderedList) return
 
         resetParagraphType(paragraph = paragraph)
     }
 
     fun toggleOrderedList() {
-        val paragraph = getRichParagraphByTextIndex(selection.min - 1) ?: return
-        if (paragraph.type is OrderedList) removeOrderedList()
-        else addOrderedList()
+        val firstParagraph = getRichParagraphByTextIndex(selection.min - 1) ?: return
+        val paragraphs = getRichParagraphListByTextRange(selection).toMutableList()
+        if (paragraphs.isEmpty()) {
+            paragraphs.add(firstParagraph)
+        }
+        val removeOrderedList = firstParagraph.type is OrderedList
+        paragraphs.forEach { paragraph ->
+            if (removeOrderedList) {
+                removeOrderedList(paragraph)
+            } else {
+                addOrderedList(paragraph)
+            }
+        }
     }
 
+    @Deprecated("Use toggleOrderedList() instead to add or remove ordered list")
     fun addOrderedList() {
         val paragraph = getRichParagraphByTextIndex(selection.min - 1) ?: return
+        addOrderedList(paragraph)
```

Comment:
Same as above, this shouldn't be deprecated.

Reply:
ready